src/components/select/simple-select/simple-select.test.tsx

```diff
@@ -1,85 +1,794 @@
+
+    await waitFor(() => {
+      expect(screen.queryByRole("listbox")).not.toBeInTheDocument();
+    });
+  });
+});
+
+describe("onClick prop", () => {
+  it("calls onClick callback when input is clicked", async () => {
+    const onClick = jest.fn();
+    const user = userEvent.setup();
+    render(
+      <SimpleSelect label="Colour" onChange={() => {}} onClick={onClick}>
+        <Option text="amber" value="amber" />
+      </SimpleSelect>
+    );
+
+    await user.click(screen.getByRole("combobox"));
+
+    expect(onClick).toHaveBeenCalledTimes(1);
+  });
+
+  it("does not call onClick callback when disabled input is clicked", async () => {
+    const onClick = jest.fn();
+    const user = userEvent.setup();
+    render(
+      <SimpleSelect
+        label="Colour"
         onChange={() => {}}
-        value=""
+        onClick={onClick}
+        disabled
       >
-        <Option text="Amber" value="1" />
+        <Option text="amber" value="amber" />
       </SimpleSelect>
-    </>
+    );
+
+    await user.click(screen.getByRole("combobox"));
+
+    expect(onClick).not.toHaveBeenCalled();
+  });
+
+  it("does not call onClick callback when read-only input is clicked", async () => {
+    const onClick = jest.fn();
+    const user = userEvent.setup();
+    render(
+      <SimpleSelect
+        label="Colour"
+        onChange={() => {}}
+        onClick={onClick}
+        readOnly
+      >
+        <Option text="amber" value="amber" />
+      </SimpleSelect>
+    );
+
+    await user.click(screen.getByRole("textbox"));
+
+    expect(onClick).not.toHaveBeenCalled();
+  });
+});
+
+test("calls onOpen callback, when list is opened and onOpen is provided", async () => {
+  const onOpen = jest.fn();
+  const user = userEvent.setup();
+  render(
+    <SimpleSelect label="Colour" onChange={() => {}} onOpen={onOpen}>
+      <Option text="amber" value="amber" />
+    </SimpleSelect>
   );
 
-  expect(screen.getByRole("combobox")).toHaveAccessibleName("My Select");
+  await user.click(screen.getByRole("combobox"));
+
+  expect(onOpen).toHaveBeenCalledTimes(1);
 });
+
+test("calls onKeyDown callback with information about the pressed key when typing in input", async () => {
+  const user = userEvent.setup();
+  const onKeyDown = jest.fn();
+  render(
+    <SimpleSelect label="Colour" onChange={() => {}} onKeyDown={onKeyDown}>
+      <Option text="amber" value="amber" />
+    </SimpleSelect>
+  );
+
+  await user.type(screen.getByRole("combobox"), "{c}");
+
+  expect(onKeyDown).toHaveBeenCalledWith(expect.objectContaining({ key: "c" }));
+});
+
+test("calls onFocus callback when input is focused", async () => {
+  const onFocus = jest.fn();
+  const user = userEvent.setup();
+  render(
+    <SimpleSelect label="Colour" onChange={() => {}} onFocus={onFocus}>
+      <Option text="amber" value="amber" />
+    </SimpleSelect>
+  );
+
+  await user.tab();
+
+  expect(onFocus).toHaveBeenCalledTimes(1);
+});
+
+describe("onBlur prop", () => {
```

Comment:
Similar comments above about tidying up how this reads a bit etc (again non-blocking)